lib/rules/no-empty-pattern.js

```diff
@@ -19,18 +19,57 @@ module.exports = {
             url: "https://eslint.org/docs/latest/rules/no-empty-pattern"
         },
 
-        schema: [],
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    allowObjectPatternsAsParameters: {
+                        type: "boolean",
+                        default: false
+                    }
+                },
+                additionalProperties: false
+            }
+        ],
 
         messages: {
             unexpected: "Unexpected empty {{type}} pattern."
         }
     },
 
     create(context) {
+        const options = context.options[0] || {},
+            allowObjectPatternsAsParameters = options.allowObjectPatternsAsParameters || false;
+
         return {
             ObjectPattern(node) {
-                if (node.properties.length === 0) {
-                    context.report({ node, messageId: "unexpected", data: { type: "object" } });
+
+                // Allow {} empty object patterns in Arrow functions as parameters when allowObjectPatternsAsParameters is true
+                if (node.parent.type === "ArrowFunctionExpression" &&
+                allowObjectPatternsAsParameters === true &&
+                node.properties.length === 0) {
+                    return;
+                }
+
+                // Allow {} = {} empty object patterns in Arrow functions as parameters when allowObjectPatternsAsParameters is true
+                if (node.parent.type === "AssignmentPattern" &&
+                node.parent.parent.type === "ArrowFunctionExpression" &&
+                node.parent.right.type === "ObjectExpression" &&
+                node.parent.right.properties.length === 0 &&
+                allowObjectPatternsAsParameters === true &&
+                node.properties.length === 0) {
+                    return;
+                }
+
+                if ((node.parent.type === "ArrowFunctionExpression" &&
+                allowObjectPatternsAsParameters === false &&
+                node.properties.length === 0) ||
```

Comment:
These conditions are redundant with the previous checks. At this point, this can be just `if (node.properties.length === 0) { contex.report(...); }`.

Or even better, we could have an early exit for non-empty patterns:

```js
ObjectPattern(node) {
    if (node.properties.length > 0) {
        return;
    }

    // check and exit if allowObjectPatternsAsParameters applies

    context.report({ node, messageId: "unexpected", data: { type: "object" } });
}
```